test/postgres-conflict.test.ts

```diff
@@ -0,0 +1,186 @@
+        expect(callbackSpy).toBeCalledTimes(2);
+        expect(results.map(r => r.newState.global.accounts)).toEqual([
+            [
+                {
+                    address: '0x123',
+                    lineItems: [],
+                    name: ''
+                }
+            ],
+            [
+                {
+                    address: '0x123',
+                    lineItems: [],
+                    name: ''
+                }
+            ]
+        ]);
+    });
+
+    it('should have conflict on operations for the same document', async ({
+        expect
+    }) => {
+        await storage.createDocument(
+            DRIVE_ID,
+            'doc3',
+            Budget.utils.createDocument(),
+            [{ syncId: '3', scope: 'global', branch: 'main' }]
+        );
+
+        let accountId = 1;
+        async function callback(documentStorage: DocumentStorage): Promise<{
+            operations: Operation[];
+            header: DocumentHeader;
+            newState?: State<any, any> | undefined;
+        }> {
+            const oldDocument = documentStorage.operations.global.reduce(
+                (doc, op) =>
+                    Budget.reducer(
+                        doc,
+                        op as unknown as Budget.BudgetStatementAction
+                    ),
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+                Budget.utils.createDocument(documentStorage as any)
+            );
+            const { document, operation } = buildOperationAndDocument(
+                Budget.reducer,
+                oldDocument,
+                Budget.actions.addAccount({ address: accountId.toString() }),
+                documentStorage.operations.global.length
+            );
+            accountId++;
+            operation.id = generateUUID();
+            operation.resultingState = document.state[operation.scope];
+            return Promise.resolve({
+                header: document,
+                operations: [operation],
+                newState: document.state
+            });
+        }
+
+        const callbackSpy = vi.fn(callback);
+
+        let prismaError: Prisma.LogEvent | undefined;
+        client.$on('error', error => {
+            prismaError = error;
+        });
+
+        const results = await Promise.all([
+            storage.addDocumentOperationsWithTransaction(
+                DRIVE_ID,
+                'doc3',
+                callbackSpy
+            ),
+            storage.addDocumentOperationsWithTransaction(
+                DRIVE_ID,
+                'doc3',
+                callbackSpy
+            )
+        ]);
+
+        expect(prismaError).toBeDefined();
+        expect(
+            prismaError?.message.includes(
+                'An operation failed because it depends on one or more records that were required but not found.'
+            )
+        ).toBe(true);
+        expect(callbackSpy).toBeCalledTimes(3);
```

Comment:
The retry error is thrown but both operations were processed successfully. CallbackSpy, which does the shuffling logic, is called 3 times as expected, since there was one retry.